src/Agent.Worker/JobRunner.cs

```diff
@@ -334,6 +334,25 @@ public async Task<TaskResult> RunAsync(Pipelines.AgentJobRequestMessage message,
                     return await CompleteJobAsync(jobServer, jobContext, message, TaskResult.Failed);
                 }
 
+                // check if knob DrainQueuesAfterTask has correct value
+                bool defaultDrainQueuesAfterTask = AgentKnobs.DrainQueuesAfterTask.GetValue<BuiltInDefaultKnobSource>(jobContext).AsBoolean();
+                try
+                {
+                    AgentKnobs.DrainQueuesAfterTask.GetValue(jobContext).AsBooleanStrict();
+                }
+                catch (Exception ex)
+                {
+                    switch (ex)
+                    {
+                        case ArgumentException _:
+                        case FormatException _:
+                            jobContext.Warning($"Knob 'AGENT_DRAIN_QUEUES_AFTER_TASK' has wrong value, default value '{defaultDrainQueuesAfterTask}' will be set");
+                            break;
+                        default:
+                            throw;
```

Comment:
throw ex? or it's do the same?

Reply:
"throw;" (without the ex) is preferred as it will preserve the original call stack; the later will overwrite the callstack.  Experiment with it to see the difference.